src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

```diff
@@ -0,0 +1,691 @@
+import javax.swing.event.ListSelectionEvent;
+import javax.swing.event.ListSelectionListener;
+import javax.swing.table.DefaultTableCellRenderer;
+import javax.swing.table.DefaultTableModel;
+
+import org.isf.generaldata.MessageBundle;
+import org.isf.medicalinventory.manager.MedicalInventoryManager;
+import org.isf.medicalinventory.manager.MedicalInventoryRowManager;
+import org.isf.medicalinventory.model.MedicalInventory;
+import org.isf.medicalinventory.model.MedicalInventoryRow;
+import org.isf.menu.manager.Context;
+import org.isf.utils.jobjects.GoodDateChooser;
+import org.isf.utils.jobjects.ModalJFrame;
+import org.isf.utils.jobjects.TextPrompt;
+import org.isf.utils.jobjects.TextPrompt.Show;
+import org.isf.utils.time.TimeTools;
+
+public class InventoryEdit extends ModalJFrame {
+
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = 1L;
+	
+	private static EventListenerList InventoryListeners = new EventListenerList();
+	public interface InventoryListener extends EventListener {
+		public void InventoryUpdated(AWTEvent e);
+
+		public void InventoryInserted(AWTEvent e);
+
+		public void InventoryValidated(AWTEvent e);
+
+		public void InventoryCancelled(AWTEvent e);
+	}
+
+	public static void addInventoryListener(InventoryListener l) {
+		InventoryListeners.add(InventoryListener.class, l);
+	}
+
+	private GoodDateChooser jCalendarTo;
+	private GoodDateChooser jCalendarInventory;
+	private LocalDateTime dateInventory = TimeTools.getServerDateTime();
+	private JLabel jLabelTo;
+	private JPanel panelHeader;
+	private JPanel panelFooter;
+	private JPanel panelContent;
+	private JButton closeButton;
+	private JScrollPane scrollPaneInventory;
+	private JTable jTableInventoryRow;
+	private List<MedicalInventoryRow> inventoryRowList;
+	private List<MedicalInventoryRow> inventoryRowSearchList;
+	private String[] pColums = { MessageBundle.getMessage("angal.common.code.txt"),
+			MessageBundle.getMessage("angal.medicalstockward.patient.drug.col"),
+			MessageBundle.getMessage("angal.wardpharmacy.lotnumber.col"),
+			MessageBundle.getMessage("angal.medicalstock.duedate.col"),
+			MessageBundle.getMessage("angal.inventoryrow.theorticqty.col"),
+			MessageBundle.getMessage("angal.inventoryrow.realqty.col") };
+	private int[] pColumwidth = { 100, 300, 100, 100, 100, 100 };
+	private boolean[] columnEditable = { false, false, false, false, false, true };
+	private boolean[] columnEditableView = { false, false, false, false, false, false };
+	private MedicalInventory inventory = null;
+	private JRadioButton specificRadio;
+	private JRadioButton allRadio;
+	private JTextField searchTextField;
+	private JLabel dateInventoryLabel;
+	private JTextField codeTextField;
+	private String code = null;
+	private String mode = null;
+	private JLabel referenceLabel;
+	private JTextField referenceTextField;
+	private JTextField jTetFieldEditor;
+	private JLabel loaderLabel;
+
+	private JButton moreData;
+	private int MAX_COUNT = 30;
```

Comment:
Where are these 3 next variables used?

Reply:
@ArnaudFonzam maybe you should add only what is needed at each step, to ease the reviews

Reply:
okay